sdks/java/nexmark/src/test/java/org/apache/beam/sdk/nexmark/model/sql/adapter/ModelFieldsAdapterTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.nexmark.model.sql.adapter;
+
+import static org.junit.Assert.assertSame;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.beam.sdk.extensions.sql.BeamRecordSqlType;
+
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link ModelFieldsAdapter}.
+ */
+public class ModelFieldsAdapterTest {
+
+  private static class ConcreteAdapter extends ModelFieldsAdapter<String> {
+
+    ConcreteAdapter(BeamRecordSqlType recordType) {
+      super(recordType);
+    }
+
+    @Override
+    public List<Object> getFieldsValues(String model) {
+      throw new UnsupportedOperationException("Not supported");
+    }
+  }
+
+  @Test
+  public void testGetRecordType() throws Exception {
+    BeamRecordSqlType recordType =
+        BeamRecordSqlType.create(new ArrayList<String>(), new ArrayList<Integer>());
+
+    ModelFieldsAdapter<String> adapter = new ConcreteAdapter(recordType);
+
+    assertSame(recordType, adapter.getRecordType());
```

Comment:
<!--thread_id:cc_151144510_t; commit:050bdab2afe544b8a5c58bfe1dfe1914a9820c8f; resolved:0-->
<!--section:context-quote-->
> **echauchot** wrote:
> I don't understand what it tests: the only thing that I see it testing is that `recordType` field is properly assigned in `ConcreteAdapter` 's superclass constructor. Did I miss something?

<!--section:body-->
That is pretty much what happens. It tests that record type is assigned and returned as expected.

Reply:
Then, I don't think it is very useful to test line   `ModelFieldsAdapter(BeamRecordSqlType recordType) {this.recordType = recordType; }` of ModelFieldsAdapter class.

Reply:
IMHO, I would remove this test